modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/Diagram.js

```diff
@@ -64,48 +70,109 @@ const Diagram = ({
 		radius: newPinSettings.defaultRadius,
 	});
 
-	useEffect(() => {
-		fetch(`${pinsEndpoint}${productId}/pins`)
+	const loadPins = useCallback(
+		() =>
+			fetch(`${pinsEndpoint}${PRODUCTS}/${productId}/${PINS}`, {
+				headers: new Headers({
+					Accept: 'application/json',
+					'Content-Type': 'application/json',
+				}),
+			})
+				.then((response) => response.json())
+				.then((jsonResponse) => {
+					const loadedPins = jsonResponse.items.map((item) => ({
+						cx: item.positionX,
+						cy: item.positionY,
+						id: item.id,
+						label: item.number,
+					}));
+
+					setCpins(loadedPins);
+				}),
+		[pinsEndpoint, productId]
+	);
+
+	const deletePin = (node) => {
+		fetch(`${pinsEndpoint}${PINS}/${node.id}`, {
+			headers: new Headers({
+				Accept: 'application/json',
+				'Content-Type': 'application/json',
+			}),
+			method: 'DELETE',
+		});
+
+		// toast "pins deleted"
```

Comment:
We shouldn't be committing comments like this.

Reply:
Yes i know, that would be more a question. Should we add toasts to get responses? I didn't see them in the designs but sometimes i noticed those notification and i guess to be able to use it if can be a right move.

Reply:
@garofalog that sort of decision should be handled from the product team. There are typically alerts for actions, but I would defer to your product team to make that decision.